### PR TITLE
 feat(module:progress): support gradient color, circle type support success percent

### PR DESCRIPTION
**Pull request by wendzhue**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/progress/nz-progress.utils.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+import { NzProgressColorGradient, NzProgressGradientProgress } from './nz-progress.definitions';
+
+export const sortGradient = (gradients: NzProgressGradientProgress) => {
+  let tempArr: Array<{ key: number; value: string }> = [];
+  Object.keys(gradients).forEach((key: string) => {
```
Comment (hsuanxyz) [inline components/progress/nz-progress.utils.ts]: 这要改成 `for` 不然下面的 `return` 无法退出。

**Comment by wendzhue** [comment]
Works with #3932. This CSS class `ant-progress-path-circle` is modified.

**Comment by wendzhue** [comment]
Sync styles from: https://github.com/ant-design/ant-design/pull/18284.

File: components/progress/nz-progress.component.ts [hunk 2]
```diff
@@ -31,114 +38,123 @@ export type NzProgressStrokeLinecapType = 'round' | 'square';
...
-    }
-
     if (nzStatus) {
       this.cachedStatus = this.nzStatus || this.cachedStatus;
-      this.updateIcon();
     }
 
     if (nzPercent || nzSuccessPercent) {
       const fillAll = parseInt(this.nzPercent.toString(), 10) >= 100;
+      console.log(this.nzSuccessPercent);
```
Comment (hsuanxyz) [inline components/progress/nz-progress.component.ts]:
```suggestion
```

**Pull request merged**